src/pydata_sphinx_theme/assets/styles/components/_search.scss

```diff
@@ -53,6 +53,7 @@
 
 .form-control {
   background-color: var(--pst-color-background);
+  color: var(--pst-color-text-muted);
```

Comment:
using this color, the placeholder and the written text will have the exact same color, any reason why you didn't used the default `--pst-color-text-base` ?